sources/platform/actors/development/actor_definition/dynamic_actor_memory/index.md

```diff
@@ -0,0 +1,193 @@
+---
+title: Dynamic Actor memory
+description: Learn how to automatically adjust your Actor's memory based on input size and run options, so you can optimize performance and reduce costs without manual configuration..
+slug: /actors/development/actor-definition/dynamic-actor-memory
+---
+
+**Learn how to automatically adjust your Actor's memory based on input size and run options, so you can optimize performance and reduce costs without manual configuration.**
+
+---
+
+Dynamic Actor memory allows Actors to _automatically adjust its memory allocation based on the input and run options_. Instead of always using a fixed memory value, Actor can use just the right amount of memory for each run.
+
+This helps:
+
+- _Optimize performance_ for large inputs (more memory for bigger tasks).
+- _Reduce costs_ for small runs (less memory when it’s not needed).
+- _Provide better user experience_, so users get optimal performance without having to manually configure memory.
+
+:::info
+_Important_: The memory calculated by dynamic expression is used as the default memory for the run. Users can still override it manually for each run.
+:::
+
+---
+
```

Comment:
(opt) I'm kind of missing here a section _How dynamic memory works_.

First of all, we should double-down on the fact that _dynamic_ does not mean that the Actor can on-the-fly ask for more memory. IMHO this is easily confused (we have seen it even internally).

Second, it should explain the whole "memory configuration pipeline", as in, _how is memory for a run determined_ (memory expression vs Actor defaults vs run overrides).